src/test/java/kitchenpos/application/MenuServiceTest.java

```diff
@@ -0,0 +1,372 @@
+import static kitchenpos.application.MenuProductFixture.콜라_수량_오류;
+import static kitchenpos.application.ProductFixture.맛초킹;
+import static kitchenpos.application.ProductFixture.콜라;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.junit.jupiter.api.Assertions.assertAll;
+
+import java.math.BigDecimal;
+import java.util.Arrays;
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.UUID;
+import kitchenpos.domain.Menu;
+import kitchenpos.domain.MenuGroup;
+import kitchenpos.domain.MenuGroupRepository;
+import kitchenpos.domain.MenuProduct;
+import kitchenpos.domain.MenuRepository;
+import kitchenpos.domain.Product;
+import kitchenpos.domain.ProductRepository;
+import kitchenpos.infra.ProfanityClient;
+import org.assertj.core.api.ThrowableAssert.ThrowingCallable;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.NullSource;
+import org.junit.jupiter.params.provider.ValueSource;
+
+@DisplayName("메뉴 관리")
+class MenuServiceTest {
+
+    private final MenuRepository menuRepository = new InMemoryMenuRepository();
+    private final MenuGroupRepository menuGroupRepository = new InMemoryMenuGroupRepository();
+    private final ProductRepository productRepository = new InMemoryProductRepository();
+    private final ProfanityClient profanityClient = new FakeProfanityClient();
+
+    private MenuService menuService;
+
+    @BeforeEach
+    void setUp() {
+        menuService = new MenuService(menuRepository, menuGroupRepository, productRepository, profanityClient);
+    }
+
+    @DisplayName("메뉴의 가격은 0원 이상이어야 한다.")
+    @ParameterizedTest
+    @ValueSource(strings = {"-1"})
+    @NullSource
+    void createPriceUnderZeroException(BigDecimal price) {
+        //given
+        Menu 잘못된_가격의_메뉴 = 메뉴생성()
+            .withPrice(price)
+            .build();
+
+        //when
+        ThrowingCallable actual = () -> menuService.create(잘못된_가격의_메뉴);
+
+        //then
+        assertThatThrownBy(actual).isInstanceOf(IllegalArgumentException.class);
+    }
+
+    @DisplayName("메뉴는 메뉴 그룹에 속해야 한다.")
+    @Test
+    void createMenuGroupNotExistException() {
+        //given
+        Menu 메뉴_그룹에_속하지_않은_메뉴 = 메뉴생성()
+            .withMenuGroup(null)
+            .build();
+
+        //when
+        ThrowingCallable actual = () -> menuService.create(메뉴_그룹에_속하지_않은_메뉴);
+
+        //then
+        assertThatThrownBy(actual).isInstanceOf(NoSuchElementException.class);
+    }
+
+    @DisplayName("메뉴는 1개 이상의 상품을 포함해야 한다.")
+    @Test
+    void createMenuProductsNotExistException() {
+        //given
+        MenuGroup 세트_메뉴 = menuGroupRepository.save(세트메뉴);
+
+        Menu 상품_없는_메뉴 = 메뉴생성()
+            .withMenuGroup(세트_메뉴)
+            .withMenuGroupId(세트_메뉴.getId())
```

Comment:
아래 코멘트에 대해 고민해 봐요.
https://github.com/next-step/ddd-legacy/pull/258#discussion_r835986814

```suggestion
            .withMenuGroupId(세트_메뉴.getId())
            .withMenuProducts(Collections.emptyList())
```